libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UIViewModel.kt

```diff
@@ -27,7 +27,7 @@ abstract class UIViewModel<State, Action>(initialState: State) : MapboxNavigatio
 
     private val _action = MutableSharedFlow<Action>(extraBufferCapacity = 1)
     val action: Flow<Action> = _action
-    private val _state: MutableStateFlow<State> = MutableStateFlow(initialState)
+    protected val _state: MutableStateFlow<State> = MutableStateFlow(initialState)
```

Comment:
I understand the problem and the reason behind this change. This is the 2nd time we have come across this problem, that a ViewModel is detached while an action is still processing, first time being [here](https://github.com/mapbox/navigation-sdks/issues/1530#issuecomment-1067221267). I would want us to avoid opening the state so that other view models could just update it from the outside. 

However, that is another problem not in the scope of this PR. Let's merge this now, but I will create a follow up ticket that could ensure that a view model does not detach if an action is currently under process.

cc @tomaszrybakiewicz

Reply:
Making it `protected` doesn't open the state for others. It only opens it for subclasses that already own that state.

The **Store** I introduce in [this PR](https://github.com/mapbox/mapbox-navigation-android/pull/5626) solves this problem. It's coroutine scope matches `DropInNavigationViewModel` lifecycle and is not canceled by MapboxNavigation onDetached.

Reply:
We could also resolve this with synchronous actions https://github.com/mapbox/navigation-sdks/issues/1571

The centralized **Store** approach still has some issues to resolve before we can move forward with it